### PR TITLE
Upgrade pipenv to 2022.9.24

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Pipenv.kt [hunk 1]
```diff
@@ -72,7 +73,11 @@ class Pipenv(
 
         logger.info { "Generating '${requirementsFile.name}' file in '$workingDir' directory..." }
 
-        val requirements = ProcessCapture(workingDir, command(), "lock", "--requirements").requireSuccess().stdout
+        val requirements = if (Semver(getVersion()) >= Semver("2022.8.13")) {
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pipenv.kt:76]: Should we maybe guard against `getVersion()` not being parsable as a `Semver` here? and maybe extract `Semver("2022.8.13")` to a class variable with a speaking name?

File: analyzer/src/main/kotlin/managers/Pipenv.kt [hunk 2]
```diff
@@ -72,7 +73,11 @@ class Pipenv(
...
+            ProcessCapture(workingDir, command(), "requirements").requireSuccess().stdout
+        } else {
+            ProcessCapture(workingDir, command(), "lock", "--requirements").requireSuccess().stdout
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pipenv.kt:79]: Nit: Maybe pull out `.requireSuccess().stdout` of the ` if`?

File: analyzer/src/main/kotlin/managers/Pipenv.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Pipenv.kt]: Did you intend to also address the point from my first sentence?

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Pipenv.kt]: I forgot to respond, no, all versions of Pipenv so far are semver compliant so I don't think this is necessary.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**